python/tk_framework_desktopserver/settings.py

```diff
@@ -35,122 +34,97 @@ class Settings(object):
-                config, self._PORT_SETTING, int
-            )
-            certificate_folder = self._get_value(
-                config, self._CERTIFICATE_FOLDER_SETTING
-            )
-            integration_enabled = self._get_value(
-                config, self._ENABLED
-            )
-        else:
-            from sgtk.util import UserSettings
-            user_settings = UserSettings()
-            port = user_settings.get_integer_setting(
-                self._BROWSER_INTEGRATION, self._PORT_SETTING
-            )
-            certificate_folder = user_settings.get_setting(
-                self._BROWSER_INTEGRATION, self._CERTIFICATE_FOLDER_SETTING
-            )
-            integration_enabled = UserSettings().get_boolean_setting(self._BROWSER_INTEGRATION, self._ENABLED)
+        from sgtk.util import UserSettings
+        user_settings = UserSettings()
+        port = user_settings.get_integer_setting(
+            self._BROWSER_INTEGRATION, self._PORT_SETTING
+        )
+        certificate_folder = user_settings.get_setting(
+            self._BROWSER_INTEGRATION, self._CERTIFICATE_FOLDER_SETTING
+        )
+        integration_enabled = UserSettings().get_boolean_setting(self._BROWSER_INTEGRATION, self._ENABLED)
+
+        raw_host_aliases = {}
+        if UserSettings().get_section_settings(self._HOST_ALIASES):
+            raw_host_aliases = {
+                name: UserSettings().get_setting(
+                    self._HOST_ALIASES, name
+                )
+                for name in UserSettings().get_section_settings(self._HOST_ALIASES)
+            }
 
         self._port = port or self._DEFAULT_PORT
         self._certificate_folder = certificate_folder or self._default_certificate_folder
         self._integration_enabled = integration_enabled
 
-    def _load_config(self, path):
-        """
-        Loads the configuration at a given location and returns it.
-
-        :param path: Path to the configuration to load.
-
-        :returns: A ConfigParser instance with the contents from the configuration file.
-        """
-        config = ConfigParser.SafeConfigParser()
-        if os.path.exists(path):
-            config.read(path)
-        return config
+        # Keep the raw aliases for support, but filter the settings for API users.
+        self._raw_host_aliases = raw_host_aliases
+        self._host_aliases = {}
+        # Ensure we have a string, and then split the string on commas, make sure we're stripping
+        # out beginning and end of string whitespaces and then lowercase everything.
+        # Also skip empty tokens.
+        for main_host, secondary_hosts in raw_host_aliases.iteritems():
+            main_host = main_host.strip().lower()
+            # Skip empty hosts.
+            if not main_host:
+                continue
+
+            self._host_aliases[main_host] = [
+                secondary_host.lower().strip()
+                for secondary_host in secondary_hosts.split(",")
```

Comment:
nitpick: might could do this in a single go with re.split splitting on r"[,\s]+"

Reply:
I guess we could do that, but we would still require the list comprehension to put anything in lower case. I feel like invoking strip() is clearer to the reader than a regex however.